### ISSUE TITLE
detect existing KeyPair to avoid failure on second cluster creation

### ISSUE THREAD

**Issue by jstrachan** [weaveworks/eksctl#91]
I created a cluster and it worked great. I zapped it via the CloudFormation console.

Then the next time I tried to create a cluster I get an error:
```
2018-07-03T12:07:39+01:00 [ℹ]  importing SSH public key "/Users/jstrachan/.ssh/id_jstrachan_testing_rsa.pub" as "EKS-eks1"
2018-07-03T12:07:40+01:00 [✖]  importing SSH public key: InvalidKeyPair.Duplicate: The keypair 'EKS-eks1' already exists.
```
I had to manually work around this via:
```
aws ec2 delete-key-pair --key-name EKS-eks1 --region us-west-2
```
It'd be nice to detect this error message and log as a warning and continue; or better use the `aws` CLI to query if the KeyPair exists for the given name first?

**Comment by errordeveloper** [comment]
> It'd be nice to detect this error message and log as a warning and continue; or better use the aws CLI to query if the KeyPair exists for the given name first?

Something like that... but probably better to add unique name suffix based on checksum, and when such name exists, double check fingerprint, and error if it's absolutely not the same thing, but continue otherwise.

Also, if you want to make use of existing key, you can just pass `--ssh-public-key EKS-eks1`.

https://github.com/weaveworks/eksctl/blob/a71ba696af68b6656fccaa8e49e8f7c15715cee1/pkg/eks/auth.go#L26-L63

**Comment by errordeveloper** [comment]
I think checking the fingerprint should be enough really, will take a stab at it.

**Comment by errordeveloper** [comment]
Arguably https://github.com/weaveworks/eksctl/issues/105 is another way to fix this, but #107 is a plausible start (and needed anyway).

**Comment by errordeveloper** [comment]
Arguably #105 is another way to fix this, but #106 is a plausible start (and needed anyway).
